### PR TITLE
Showif

### LINKED ISSUE #10: Include showIf option
include a showIf option. This means only show the given question if the condition matches or callback returns true

### PR DESCRIPTION
**Pull request by awebdeveloper**
Implements showIf callback method. Should fix #10

### CODE REVIEW

File: oncogene.js [hunk 1]
```diff
@@ -22,7 +22,14 @@ class Oncogene {
...
         this.stepInx = Math.max(0, this.stepInx + stepSize)
 
-        if (this.stepInx >= this.steps.length) return this.renderResult()
+        if (this.stepInx >= this.steps.length) {
+            return this.renderResult()
+        }
+
+        const step = this.getStep();
+        if (this.constructor.isFunction(step.showIf) && !step.showIf(this.config)) {
+           return this.nextStep();
```
Comment (gwer) [inline oncogene.js]: stepSize may be negative. So we need `this.nextStep(stepSize)` instead of `this.nextStep()`.

**Review by gwer — CHANGES_REQUESTED** [review]

File: oncogene.js (hunk 1 above)
Comment (awebdeveloper) [reply oncogene.js]: Sorry can you explain why

Comment (awebdeveloper) [reply oncogene.js]: nextStep has no parameter

Comment (gwer) [reply oncogene.js]: I'm sorry, `this.makeStep(stepSize)` of course.

Comment (awebdeveloper) [reply oncogene.js]: sorry i still dont get it

Comment (gwer) [reply oncogene.js]: `this.nextStep()` is an alias for `this.makeStep(1)`. Oncogene's API allows you to call `this.makeStep(-1)` or `this.prevStep()`.

If you want to go back to the previous step, and the previous step shouldn't be showed, you should go back, not forward.

**Comment by awebdeveloper** [comment]
done

**Comment by gwer** [comment]
Thanks

**Pull request merged**